### ISSUE TITLE
Video not working

### ISSUE THREAD

**Issue by casualGamer-dev** [chirag-droid/animeflix#138]
Unhandled Runtime Error
TypeError: Cannot read properties of undefined (reading '0')

Source
src\pages\watch\[id].tsx (108:17) @ eval

  106 |     if (isLoading) return;
  107 | 
> 108 |     if (!sources[0].file.match(proxyFreeUrls)) {
      |                 ^
  109 |       setProxy(true);
  110 |     }
  111 |   }, [isLoading, setProxy, sources]);

**Comment by chirag-droid** [comment]
Yeah that pull request currently has some bugs that needs to be fixed before I can merge it.

Can you provide some more info regarding what anime you were watching and what episode.

**Comment by casualGamer-dev** [comment]
i have tried watching a coupe of animes and different episodes. all read to an error. I have managed to narrow it down to 
line  19 of gogoganime.ts

**Comment by chirag-droid** [comment]
thanks. will look into it

**Comment by AshuYadavYT** [comment]
> Yeah that pull request currently has some bugs that needs to be fixed before I can merge it.
> 
> Can you provide some more info regarding what anime you were watching and what episode.

Komi can't communicate season 1 & 2 eps
Overflow all eps
etc. 
![image](https://user-images.githubusercontent.com/90438479/169689662-d5cb2e3c-86cb-4561-bde1-001d2fc4e4b1.png)
this loading loop run forever

**Comment by casualGamer-dev** [comment]
@chirag-droid  can u share the version that hosted live on netlify atm ?

**Comment by chirag-droid** [comment]
https://animeflixxx.netlify.com

https://animeflix.up.railway.app

These changes haven't been merged with main branch.

I will look into these issues when I get time

**Comment by chirag-droid** [comment]
> > Yeah that pull request currently has some bugs that needs to be fixed before I can merge it.
> > Can you provide some more info regarding what anime you were watching and what episode.
> 
> Komi can't communicate season 1 & 2 eps Overflow all eps etc. ![image](https://user-images.githubusercontent.com/90438479/169689662-d5cb2e3c-86cb-4561-bde1-001d2fc4e4b1.png) this loading loop run forever

@AshuYadavYT I have fixed it now. And adult content is not available on the site.

It was a very minor error that could be prevented with proper testing. I have pushed the fix to upstream branch. Rewrite branch will still show error

**Comment by casualGamer-dev** [comment]
so what was the issue

**Comment by chirag-droid** [comment]
So for some search the english title works and for some romaji. So we parallely fetch the data by search both the titles.

For this particular anime, the english data was empty and romaji one was there. See 7cfca0ec408ab8a68a7327991d4b450aef547679

**Comment by casualGamer-dev** [comment]
I bring to you a new bug. Click on watch dub gives an client side error

**Comment by casualGamer-dev** [comment]
I have also found Another issue . Neon genesis plays sub by default. and clicking on dub gives an error

**Comment by casualGamer-dev** [comment]
I have narrowed down the problem. Only dub exists but sub doesnt
....
